hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/debug/replicas/BlockVerificationResult.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hadoop.ozone.debug.replicas;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * Json structure for replicas to pass through each check and give output.
+ */
+public class BlockVerificationResult {
+
+  private final boolean pass;
+  private final List<FailureDetail> failures;
+
+  public BlockVerificationResult(boolean pass, List<FailureDetail> failures) {
+    this.pass = pass;
+    this.failures = failures;
+  }
+
+  public static BlockVerificationResult pass() {
+    return new BlockVerificationResult(true, null);
+  }
+
+  public static BlockVerificationResult failCheck(String message) {
+    return new BlockVerificationResult(false,
+        Collections.singletonList(new FailureDetail(true, message)));
+  }
+
+  public static BlockVerificationResult failIncomplete(String message) {
+    return new BlockVerificationResult(false,
+        Collections.singletonList(new FailureDetail(false, message)));
+  }
+
+  public boolean passed() {
+    return pass;
+  }
+
+  public Optional<List<FailureDetail>> getFailures() {
+    return Optional.ofNullable(failures);
+  }
+
+  /**
+   * Details about the check failure.
+   */
+  public static class FailureDetail {
+    // indicates whether the check finished and failed,
+    // or it was unable to finish due to connection or other issues
+    private final boolean completed;
+    private final String message;
+
+    public FailureDetail(boolean completed, String message) {
+      this.completed = completed;
```

Comment:
Now that I'm looking at the whole output it looks like `complete` should actually be a top level property of the `BlockVerificationResult`. Then the failures can just be returned as a list of strings for now. When rendering it as json we can keep it as an object with a `message` field to keep the output format extensible later. This internal format is easy enough to expand later if just a list of strings is not enough.